modules/steam/network.nix

```diff
@@ -0,0 +1,24 @@
+{lib, config, pkgs, ...}: let
+  cfg = config.jovian.steam;
+in with lib; {
+  options.jovian.steam.network = {
+    enable = mkOption {
+      description = mdDoc "Open required ports for various Steam features.";
+      type = types.bool;
+      default = cfg.enable;
+    };
+  };
+  config = mkIf cfg.network.enable {
+    networking.firewall = {
+      allowedTCPPorts = [ 24070 ];
+      allowedUDPPorts = [ 3478 ];
+      allowedTCPPortRanges= [
+        {from= 27015; to= 27050;}
+      ];
+      allowedUDPPortRanges= [
+        {from=4379; to=4380;}
+        {from=27000; to=27100;}
+      ];
```

Comment:
Can you add a comment describing what each range does, and the reference link?

That would help future readers who'll end-up reading that determine why, and what to add if needed.

Also, coding style, always put spaces within the braces `{ ... }`, and around ` = `. One-liner attrsets are fine for small "value-like" attrsets like those though.